include/robot_localization/filter_base.h

```diff
@@ -282,6 +288,18 @@ class FilterBase
     //!
     void setDebug(const bool debug, std::ostream *outStream = NULL);
 
+    //! @brief Sets the filter into verbose mode
+    //!
+    //! NOTE: this will generates a lot of debug output to the provided stream.
+    //! The value must be a pointer to a valid ostream object.
+    //!
+    //! @param[in] verbose - Whether or not to place the filter in verbose mode
+    //! @param[in] outStream - If verbose is true, then this must have a valid pointer.
+    //! If the pointer is invalid, the filter will not enter verbose mode. If verbose is
+    //! false, outStream is ignored.
+    //!
+    void setVerbose(const bool verbose, std::ostream *outStream = NULL);
```

Comment:
Just an opinion here, it's usually safer to use `nullptr` instead of `NULL`. [Pretty good reasoning here](https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced). Not sure we're really worried about overloads here, but it's something to consider.

Reply:
In our codebase, definitely. I would only retain `NULL` here if stylistic consistency comes into play.

Reply:
Has been updated in the commit below.